allure-testng/src/test/java/io/qameta/allure/testng/samples/ParameterizedTest.java

```diff
@@ -38,13 +40,36 @@ public Object[][] testData() {
         };
     }
 
+    @DataProvider
+    public static Object[][] testDataForParamNames() {
+        return new Object[][]{
+            {1, 1, 2, 5},
+            {2, 2, 4, 5}
+        };
+    }
+
     @Test(dataProvider = "testData")
     public void parameterizedTest(String param) {
         step(param);
     }
 
-    @Step
-    public void step(String param) {
+    @Test(dataProvider = "testDataForParamNames")
```

Comment:
Create a separate sample, and don't forget to add the test that will verify the data, see https://github.com/allure-framework/allure-java/blob/main/allure-testng/src/test/java/io/qameta/allure/testng/AllureTestNgTest.java#L1111-L1124 as an example

Reply:
Addressed, thanks @baev.